### PR TITLE
[Merged by Bors] - chore: generalised some results from ring to semiring

### PR DESCRIPTION
**Pull request by XavierXarles**
Changed the hypothesis of some results from ring to semiring, without changing the proofs. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ericrbg** [comment]
sorry, what are these random files?

**Review by ericrbg — APPROVED** [review]
maintainer merge

File: Mathlib/RingTheory/Ideal/Basic.lean [hunk 1]
```diff
@@ -542,14 +542,14 @@ instance (priority := 100) IsMaximal.isPrime' (I : Ideal α) : ∀ [_H : I.IsMax
   @IsMaximal.isPrime _ _ _
 #align ideal.is_maximal.is_prime' Ideal.IsMaximal.isPrime'
 
-theorem span_singleton_lt_span_singleton [CommRing β] [IsDomain β] {x y : β} :
-    span ({x} : Set β) < span ({y} : Set β) ↔ DvdNotUnit y x := by
+theorem span_singleton_lt_span_singleton  [IsDomain α] {x y : α} :
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Ideal/Basic.lean:545]:
```suggestion
theorem span_singleton_lt_span_singleton [IsDomain α] {x y : α} :
```

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by eric-wieser** [comment]
Note for next time that you can just bors r+ again here yourself

bors r+

**Pull request closed without merging**